src/dialogsformaction.cpp

```diff
@@ -103,29 +103,16 @@ bool DialogsFormAction::process_operation(Operation op,
 	}
 	break;
 	case OP_PREV:
-	case OP_SK_UP:
-		dialogs_list.move_up(cfg->get_configvalue_as_bool("wrap-scroll"));
+		dialogs_list.get_movement_control().move_up(cfg->get_configvalue_as_bool("wrap-scroll"));
 		break;
 	case OP_NEXT:
-	case OP_SK_DOWN:
-		dialogs_list.move_down(cfg->get_configvalue_as_bool("wrap-scroll"));
-		break;
-	case OP_SK_HOME:
-		dialogs_list.move_to_first();
-		break;
-	case OP_SK_END:
-		dialogs_list.move_to_last();
-		break;
-	case OP_SK_PGUP:
-		dialogs_list.move_page_up(cfg->get_configvalue_as_bool("wrap-scroll"));
-		break;
-	case OP_SK_PGDOWN:
-		dialogs_list.move_page_down(cfg->get_configvalue_as_bool("wrap-scroll"));
+		dialogs_list.get_movement_control().move_down(cfg->get_configvalue_as_bool("wrap-scroll"));
 		break;
 	case OP_QUIT:
 		v->pop_current_formaction();
 		break;
 	default:
+		handle_list_operations(dialogs_list.get_movement_control(), op);
```

Comment:
```suggestion
		if (handle_list_operations(dialogs_list.get_movement_control(), op)) {
			break;
		}
```

(ditto in other places that call `handle_list_operations`)

This will establish a pattern of extending the `default` clause with more such methods, and ensure that only one method handles a particular operation.

Reply:
Thanks, I like it. It makes it immediately obvious how to extend this if there is any more de-duplication effort